### ISSUE TITLE
Daily log

### ISSUE THREAD

**Issue by lucbettaieb** [lucbettaieb/thesis#9]
# Daily log
FWIW, daily logging of work goes here.

**Comment by lucbettaieb** [comment]
### 1/17/17
 - Came back to lab and got things configured again
 - Analyzed current state of project and began breaking down what needs to be done still into different sections
 - Began trying to figure out why the CNN Localizer operates slowly.  Turns out it is running the graph tremendously slowly for some reason.  When the `g_session_ptr`runs the graph and returns a status, the program halts for ~20 seconds.  Compiling for release does not change anything.

**Comment by lucbettaieb** [comment]
### 1/18/17
 - Had my meeting with Dr. Newman, hashed out deadlines and expectations for my thesis work.

**Thesis Structure**
1. Literature Review / Background (Start this now)
   - Background on **(a)** robotics, **(b)** localization, **(c)** ROS, **(d)** amcl, **(e)** kidnapped / wake-up robot problem, **(f)** place recognition, **(g)** deep learning, **(h)** artificial neural networks, **(i)** convolutional neural networks, **(j)** transfer learning, **(k)** pose estimation...
   - This section should be around twenty pages and include much literature review for each section as well as insights into the motivation of the thesis work.
2. BEEF
3. TOFU
4. CHICKEN
5. Conclusions and future work
   - Draw conclusions on what my work has accomplished and means for the research community.  Provide future insight into what should be done with respect to what I have accomplished.

 - Began collecting material and organizing everything for the starting of my paper.

**Comment by lucbettaieb** [comment]
### 1/19/17
 - Worked on debugging the slowness issue and am attributing it to ginormous tensors being passed into a retrained network of ginormous tensors.  No wonder it takes a long time.
 - Designed the datasets to be gathered.

**Comment by lucbettaieb** [comment]
Took 1/20/17 off for the inauguration of the first orange president.

### 1/23/17
#### Accomplished
 - Fixed the crappy scan being generated from depthimage_to_laserscan and pointcloud_to_laserscan by installing a real LIDAR on the robot.  The LIDAR is a Hokuyo URG-04LX-UGO1.
 - Modified the `deeplearning_bringup` on a new branch to include a transform from the `camera_link` to `laser`, the frame used by hokuyo_node.
 - Created a new package, deeplearning_launch for launch files to be used on the craptop that start the various maps + amcl.
 - Ran AMCL and the refresh rate is so much better.  Runs so much more smoothely.
#### Still need to...
 - Modify data_collector to use odom updates to snapshot images.  Maybe make a metric that must be satisfied to take pictures at different angles.
 - Still need lots of data for each image.
 - Take training set data in 640x480 and use python to generate other sets of data.
 - Create combinations and standalones and retrain the inception network with them.

**Comment by lucbettaieb** [comment]
fill in logs later
https://youtu.be/0mCX70AJZGQ

**Comment by lucbettaieb** [comment]
### 2/3/17
Haven't been updating recently because I'm lazy
 - Debugged issue in cnn_localization.  I realized that passing images into the graph is best done at a default size of 299x299.  This was discovered by looking inside label_image.cc which is in the Tensorflow examples.  I still need to experiment more with different sizing but it seems to work well.
 - Gathered new 4 class RGB data that was resized to 400x400.  Also have samples at 300, 200, 100, and 32x32.
 - Worked on the introduction chapter of my thesis quite a bit.
 - Did a few different trials with the different graphs.  Solved the slow data processing issue (got 20 seconds down to 5)

**Comment by lucbettaieb** [comment]
Wow I'm lazy.
### 2/8/17
 - Ran retrained model on turtlebot and found that it performed like crap.  Remembered DIGITS.
 - Trained models on DIGITS and found that they do very well.  May be overtraining, but does it matter?
 - Doesn't seem to be overtraining actually because it can still identify new data...  Shift invariance is accounted for and OK.
 - ~90% accuracy retraining GoogLeNet using DIGITS

### 2/9/17
 - Began exploring how to integrate the DIGITS model in my pre-existing code.
 - Created a script to convert the caffemodel / prototxt to a tensorflow pb graph with help from pre-existing code.
 - It didn't work at first, but found an open PR that adds support for exporting a pb graph.  Going to explore that option.
 - Also have ros-caffe, an open source package, as a backup.  Could be useful for bridging the two.

### 2/10/17
 - Presented to the group on my approaches to integrating ROS and TF using C++.
 - Showed DIGITS again, too.
 - Began experimenting in DIGITS

### 2/13/17
 - Began running more experiments in DIGITS
 - Found that LeNet (the old one!) is actually pretty good at taking in 32x32 RGB images, squashing to 28x28, and determining features from the small images... **~93% after 30 epochs!**
 - AlexNet needs a long time to converge, but is good, too. 
 - Need to experiment with other solvers.
